fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2182,10 +2182,15 @@ ACTOR Future<Void> commitProxyServerCore(CommitProxyInterface proxy,
 	// ((SERVER_MEM_LIMIT * COMMIT_BATCHES_MEM_FRACTION_OF_TOTAL) / COMMIT_BATCHES_MEM_TO_TOTAL_MEM_SCALE_FACTOR) is
 	// only a approximate formula for limiting the memory used. COMMIT_BATCHES_MEM_TO_TOTAL_MEM_SCALE_FACTOR is an
 	// estimate based on experiments and not an accurate one.
-	state int64_t commitBatchesMemoryLimit = std::min(
-	    SERVER_KNOBS->COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT,
-	    static_cast<int64_t>((SERVER_KNOBS->SERVER_MEM_LIMIT * SERVER_KNOBS->COMMIT_BATCHES_MEM_FRACTION_OF_TOTAL) /
-	                         SERVER_KNOBS->COMMIT_BATCHES_MEM_TO_TOTAL_MEM_SCALE_FACTOR));
+	state int64_t commitBatchesMemoryLimit = (SERVER_KNOBS->COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT > 0)
+	                                             ? SERVER_KNOBS->COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT
+	                                             : 8LL << 30;
```

Comment:
This logic seems a bit strange because it introduces a constant that wasn't there before which I guess represents "unlimited" if the knob is set to 0.  I don't think this would ever cause a problem, but it is different from the previous logic which would use 0 if COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT is 0.

Reply:
The thing is, both `SERVER_MEM_LIMIT` and `COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT` currently use `--memory` as default, which can be 0. Let me instead not setting `COMMIT_BATCHES_MEM_BYTES_HARD_LIMIT` if `--memory` is 0 and use its initial value in ServerKnobs.cpp, which is 8G.